Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp

```diff
@@ -21,6 +25,17 @@
 
 namespace AWSNativeSDKInit
 {
+    // Prevent/allow the use of the Amazon EC2 instance metadata service (IMDS). 
+    // AWS C++ SDK can reach out to EC2 metadata service for region, config or credentials, but unless code
+    // is running on EC2 compute such calls will fail and waste network resources.
+    void OnBlockEC2MetadataCalls(const bool& setMode)
+    {
+        AZ::Utils::SetEnv("AWS_EC2_METADATA_DISABLED", (setMode ? "True" : "False"), true);
```

Comment:
Minor: What if users already set the AWS_EC2_METADATA_DISABLED environment variable on their machine? May worth checking whether the variable has already been set before override it.